src/prefect/cli/deployment.py

```diff
@@ -516,12 +517,30 @@ async def run(
         None,
         "--start-at",
     ),
+    watch: bool = typer.Option(
+        None,
+        "--watch",
+        "-w",
+        help="Whether to poll the flow run until a terminal state is reached.",
+    ),
+    watch_interval: int = typer.Option(
+        None,
+        "--watch-interval",
+        help="How often to poll the flow run for state changes (in seconds).",
+    ),
+    timeout: int = typer.Option(
+        None, "--timeout", "-wt", help="Timeout for `--watch`."
+    ),
```

Comment:
Thoughts on calling this `--watch-timeout`? Otherwise I think users could interpret as the flow run timeouts after that period of time. Also, I think that we would want to be sure a user who specifies a `--watch-timeout` has passed in `--watch`